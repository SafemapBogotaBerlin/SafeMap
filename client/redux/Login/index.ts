import { createSlice } from '@reduxjs/toolkit';

const loginSlice = createSlice({
  name: 'loggedin',
  initialState: {
    value: false,
  },
  reducers: {
    login: (state) => {
      state.value = !state.value;
    },
  },
});

export const { login } = loginSlice.actions;
export default loginSlice.reducer;
