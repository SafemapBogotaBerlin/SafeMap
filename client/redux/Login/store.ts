import { configureStore } from '@reduxjs/toolkit';
import loginReducer from '.';
export const store = configureStore({
  reducer: {
    test: loginReducer,
  },
});
export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
