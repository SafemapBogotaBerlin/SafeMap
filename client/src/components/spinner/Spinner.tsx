import React from 'react';
import {ActivityIndicator, StyleSheet, View} from 'react-native';
import { styles } from './style';

const Spinner = () => (
  <View style={[styles.container, styles.horizontal]}>
    <ActivityIndicator size="large" color="##0C5A1E" testID="spinner" />
  </View>
);

export default Spinner;