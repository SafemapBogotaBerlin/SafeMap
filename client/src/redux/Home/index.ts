import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import type { PayloadAction } from '@reduxjs/toolkit';
import { Coordinates, Point, DataObject } from '../../../types/point';


interface HomeState {
  hotpoints: Coordinates[];
  data: DataObject;
  selectedPoint: Coordinates | null;
  status: 'idle' | 'pending' | 'succeeded' | 'rejected';
  error: string | null;
}
 

let data: DataObject = {};


const initialState: HomeState = {

  // this should be replaced by the call to the real time database
  hotpoints: [
    { latitude: 19.532608, longitude: -99.53209 },
    { latitude: 19.632608, longitude: -99.73209 },
    { latitude: 19.732608, longitude: -99.03 },
    {latitude: 4.820954117734985, longitude: -74.06501975735756}
  ],

  data: data,

  selectedPoint: null,
  status: 'idle',
  error: null,
};

const homeSlice = createSlice({
  name: 'home',
  initialState: initialState,
  reducers: {
    // updatePoints: (state) => {
    //   update a point might be needed later
    // },
    selectPoint: (state, action: PayloadAction<Coordinates>) => {
      state.selectedPoint = { ...action.payload };
    },
  },
});

export const { selectPoint } = homeSlice.actions;
export default homeSlice.reducer;
