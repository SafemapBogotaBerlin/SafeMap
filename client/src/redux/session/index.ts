import { createSlice } from '@reduxjs/toolkit';
import { UserData } from '../../types/index';

interface LoginState {
  authenticated: boolean;
  userData: UserData;
}

const initialState: LoginState = {
  authenticated: false,
  userData: null,
};

const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: {
    setUserData: (state, action) => {
      state.userData = action.payload;
    },

    authenticate: (state) => {
      state.authenticated = true;
    },

    updateUserName: (state, action) => {
      state.userData.name = action.payload;
    },

    logout: (state) => {
      state.authenticated = false;
      state.userData = null;
    },
  },
});

export const { setUserData, authenticate, updateUserName, logout } =
  authSlice.actions;
export default authSlice.reducer;
