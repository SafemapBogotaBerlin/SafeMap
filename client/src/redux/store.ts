import { configureStore } from '@reduxjs/toolkit';
import homeReducer from './home/index';
import authReducer from './session/index';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    auth: authReducer,
  },
});
export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
