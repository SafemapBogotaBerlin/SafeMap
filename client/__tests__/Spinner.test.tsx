import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from '../src/components/spinner/Spinner';
import { Provider } from 'react-redux';
import { store } from '../src/redux/store';

describe('<Spinner />', () => {
  let component: any;

  beforeEach(() => {
    component = renderer.create(
      <Provider store={store}>
        <Spinner />
      </Provider>
    );
  });

  it('renders  correctly', () => {
    const tree: string = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Spinner has one child', () => {
    const numberOfChildren = React.Children.count(
      component.root.props.children
    );
    expect(numberOfChildren).toBe(1);
  });

  it('Spinner component renders with correct color', () => {
    const tree = component.toJSON();

    const activityIndicator = tree.children.find(
      (child) => child.type === 'ActivityIndicator'
    );

    expect(activityIndicator.props.color).toBe('##0C5A1E');
  });
});
