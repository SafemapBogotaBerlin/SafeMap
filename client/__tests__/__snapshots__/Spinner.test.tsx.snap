// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders  correctly 1`] = `
<View
  style={
    [
      {
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    color="##0C5A1E"
    size="large"
  />
</View>
`;
